### PR TITLE
feat(dashboardsV2): Init Dashboards V2

### PR DESCRIPTION
**Pull request by dashed**
- Add view for dashboard listing (VIS-333)
- Add new empty dashboard flow (VIS-337)
- Add dashboard details view (VIS-339)

### CODE REVIEW

File: src/sentry/static/sentry/app/views/dashboardsV2/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import DashboardsV1 from 'app/views/dashboards';
+
+type Props = {
+  organization: Organization;
+};
+
+class DashboardsV2Container extends React.Component<Props> {
+  static propTypes = {
+    organization: SentryTypes.Organization.isRequired,
+  };
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/dashboardsV2/index.tsx:16]: Do we need prop types, as this code will be all typescript?

File: src/sentry/static/sentry/app/views/dashboards/overviewDashboard.jsx [hunk 2]
```diff
@@ -35,4 +38,21 @@ class OverviewDashboard extends AsyncView {
...
+  if (showDashboardV2) {
+    return <DashboardDetail {...restProps} />;
+  }
+
+  return <OverviewDashboard {...restProps} />;
+}
+
+DashboardLanding.propTypes = {
+  organization: SentryTypes.Organization,
+};
```
[the remark this answers is not in the document]
Comment (dashed) [reply src/sentry/static/sentry/app/views/dashboards/overviewDashboard.jsx:56]: I just saw the PR for the TS conversion of this file. I'll rebase.

File: src/sentry/static/sentry/app/views/dashboardsV2/controls.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    if (dashboardState === 'edit') {
+      return (
+        <ButtonBar gap={1} key="edit-controls">
+          <Button
+            onClick={e => {
+              e.preventDefault();
+              this.props.onCommit();
+            }}
+            href="#finish-editing"
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/dashboardsV2/controls.tsx:38]: Do you need a href? The button component can make vanilla HTML buttons too.

File: src/sentry/static/sentry/app/views/dashboardsV2/detail.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  onEdit = () => {
+    this.setState({
+      dashboardState: 'edit',
+    });
+  };
+
+  onCommit = () => {
+    // TODO: commit dashboard changes
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/dashboardsV2/detail.tsx:60]: There should be some code in the hackweek branch you could use.

File: src/sentry/static/sentry/app/actionCreators/dashboards.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+export function createDashboard(
+  api: Client,
+  orgId: string,
+  newDashboard: DashboardListItem
+): Promise<OrgDashboardResponse> {
+  const promise: Promise<OrgDashboardResponse> = api.requestPromise(
+    `/organizations/${orgId}/dashboards/`,
+    {
+      method: 'POST',
+      data: {title: newDashboard.title},
```
Comment (markstory) [inline src/sentry/static/sentry/app/actionCreators/dashboards.tsx:20]: Should we send the widgets as well? The dashboard could be created with an initial set of widgets as the user could make a new dashboard add some widgets and then save it.

File: src/sentry/static/sentry/app/views/dashboards/overviewDashboard.tsx [hunk 6]
```diff
@@ -56,4 +59,20 @@ class OverviewDashboard extends AsyncView<Props, State> {
...
+  if (showDashboardV2) {
+    return <DashboardDetail {...restProps} />;
+  }
+
+  return <OverviewDashboard {...restProps} />;
+}
+
+DashboardLanding.propTypes = {
+  organization: SentryTypes.Organization,
+};
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/dashboards/overviewDashboard.tsx:76]: I don't think we need proptypes, this code will all be typescript.

File: src/sentry/static/sentry/app/views/dashboardsV2/detail.tsx [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+      case 'create': {
+        if (changesDashboard) {
+          createDashboard(api, organization.slug, changesDashboard).then(
+            (newDashboard: OrgDashboardResponse) => {
+              addSuccessMessage(t('Dashboard created'));
+
+              // redirect to new dashboard
+
+              browserHistory.replace({
+                pathname: `/organizations/${organization.slug}/dashboards/${newDashboard.id}`,
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/dashboardsV2/detail.tsx:156]:
```suggestion
                pathname: `/organizations/${organization.slug}/dashboards/${newDashboard.id}/`,
```

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/actionCreators/dashboards.tsx (hunk 5 above)
Comment (dashed) [reply src/sentry/static/sentry/app/actionCreators/dashboards.tsx:20]: We should. It doesn't looks like the API endpoint for `POST /organization/:orgId/dashboards/` supports this yet.

**Pull request merged**